sql-plugin/src/main/scala/ai/rapids/cudf/CudaUtil.scala

```diff
@@ -36,6 +36,25 @@ object CudaUtil {
       CudaMemcpyKind.DEFAULT)
   }
 
+  /**
+   * Asynchronously copy from `src` buffer, starting at `srcOffset`,
+   * to a destination buffer `dst` starting at `dstOffset`,
+   * `length` bytes, in the default stream.
+   * @param src source buffer
+   * @param srcOffset source offset
+   * @param dst destination buffer
+   * @param dstOffset destination offset
+   * @param length amount to copy
+   */
+  def copyAsync(src: MemoryBuffer, srcOffset: Long, dst: MemoryBuffer,
+      dstOffset: Long, length: Long): Unit = {
+    Cuda.asyncMemcpy(
```

Comment:
Ideally this should do some bounds-checking to avoid difficult-to-debug issues if we go beyond the buffer extents. I'd actually prefer this functionality was moved to `ai.rapids.cudf.MemoryBuffer` in cudf, but it's fine for here in the interim.